smoke-tests/apps/RateLimitedSampling/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/RateLimitedSamplingIngestionSamplingDisabledTest.java

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.applicationinsights.smoketest;
+
+import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.TOMCAT_8_JAVA_11;
+import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.TOMCAT_8_JAVA_11_OPENJ9;
+import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.TOMCAT_8_JAVA_17;
+import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.TOMCAT_8_JAVA_21;
+import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.TOMCAT_8_JAVA_8;
+import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.TOMCAT_8_JAVA_8_OPENJ9;
+import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.WILDFLY_13_JAVA_8;
+import static com.microsoft.applicationinsights.smoketest.EnvironmentValue.WILDFLY_13_JAVA_8_OPENJ9;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.microsoft.applicationinsights.smoketest.schemav2.Envelope;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+@UseAgent("applicationinsights-ingestion-sampling-disabled.json")
+abstract class RateLimitedSamplingIngestionSamplingDisabledTest {
+
+  @RegisterExtension static final SmokeTestExtension testing = SmokeTestExtension.create();
+
+  @Test
+  @TargetUri(value = "/sampling", callCount = 1000)
+  void testSampling() throws Exception {
+    // give some time to collect
+    Thread.sleep(SECONDS.toMillis(5));
+
+    List<Envelope> rdEnvelopes = testing.mockedIngestion.getItemsEnvelopeDataType("RequestData");
+
+    assertThat(rdEnvelopes.size()).isEqualTo(1000);
+
+    for (Envelope rdEnvelope : rdEnvelopes) {
+      assertThat(rdEnvelope.getSampleRate()).isEqualTo(99.99f);
```

Comment:
When sampling is disabled shouldn't this be 100%?

Reply:
IIRC when it's 100 then ingestion sampling will take effect

but then again, I thought ingestion sampling only takes effect when we send no sample rate, so maybe it can be 100, I'll need to re-test this to confirm